conference/NetDevConf2020-0x14/add-xdp-on-driver.org

```diff
@@ -697,12 +695,27 @@ rx              324,235     1,194,260
 tx              0           0
 #+end_src
 
-- Start sending traffic to that interface
+- start sending traffic to that interface
 #+begin_src bash
 sudo ./pktgen_sample02_multiqueue.sh -i enp2s0 -d 192.168.200.1 -s 64 \
       -m e0:d5:5e:65:ac:83 -t4 -n0
 #+end_src
-** Slide: ndo_xdp_xmit
+** Slide: test XDP: ndo_xdp_xmit (4/4)
+- create a veth pair and move one peer to a "remote" namespace
+#+begin_src bash
+ip netns add remote
+ip link add v0 type veth peer name v1 netns remote
+#+end_src
+- run /xdp_redirect/ sample from kernel tree to redirect traffic from v0 to eth0
+  - start sending traffic from v1
+#+begin_src bash
+make M=samples/bpf -j24
+sudo ./samples/bpf/xdp_redirect v0 eth0
+
+sudo ip netns exec remote ./pktgen_sample02_multiqueue.sh -i v1 ...
```

Comment:
Will this work(?) and activate the `ndo_xdp`_xmit on eth0 driver mvneta?

The traffic will be generated inside netns remote via pktgen on veth-v1, that will **allocate SKBs**.
On veth-v0 there will run a XDP prog (for virtual driver `veth` although in veth's native mode) that will redirect out physical device eth0 (driver mvneta).

Reply:
@netoptimizer veth_xdp_rcv_skb() will create the xdp_buff from skb and xdp_redirect_kern will redirect to mvneta using tx_port map returning XDP_REDIRECT. Am I missing something?